contracts/Core/VoteManager.sol

```diff
@@ -41,90 +40,92 @@ contract VoteManager is Initializable, ACL, VoteStorage {
         bytes32 secret,
         address stakerAddress
     ) external initialized checkEpoch(epoch) {
-        uint256 thisStakerId = stakeManager.getStakerId(stakerAddress);
+        uint32 thisStakerId = stakeManager.getStakerId(stakerAddress);
         require(thisStakerId > 0, "Structs.Staker does not exist");
-        Structs.Staker memory thisStaker = stakeManager.getStaker(thisStakerId);
-        require(commitments[epoch][thisStakerId] != 0x0, "not commited or already revealed");
-        require(keccak256(abi.encodePacked(epoch, root, secret)) == commitments[epoch][thisStakerId], "incorrect secret/value");
-
-        //if revealing self
-        if (msg.sender == stakerAddress) {
-            require(parameters.getState() == parameters.reveal(), "Not reveal state");
-            require(thisStaker.stake > 0, "nonpositive stake");
-            for (uint256 i = 0; i < values.length; i++) {
-                require(MerkleProof.verify(proofs[i], root, keccak256(abi.encodePacked(values[i]))), "invalid merkle proof");
-                uint256 influence = stakeManager.getInfluence(thisStakerId);
-                votes[epoch][thisStakerId][i] = Structs.Vote(values[i], thisStaker.stake);
-                voteWeights[epoch][i][values[i]] = voteWeights[epoch][i][values[i]] + influence;
-                totalInfluenceRevealed[epoch][i] = totalInfluenceRevealed[epoch][i] + influence;
-            }
-
-            commitments[epoch][thisStakerId] = 0x0;
-            stakeManager.setStakerEpochLastRevealed(thisStakerId, epoch);
-            secrets = keccak256(abi.encodePacked(secrets, secret));
-            emit Revealed(epoch, thisStakerId, thisStaker.stake, values, block.timestamp);
-        } else {
+        require(keccak256(abi.encodePacked(epoch, values, secret)) == commitments[thisStakerId].commitmentHash, "incorrect secret/value");
+        //bounty hunter
+        if (msg.sender != stakerAddress) {
             //bounty hunter revealing someone else's secret in commit state
             require(parameters.getState() == parameters.commit(), "Not commit state");
-            commitments[epoch][thisStakerId] = 0x0;
+            commitments[thisStakerId].commitmentHash = 0x0;
             stakeManager.slash(thisStakerId, msg.sender, epoch);
+            return;
         }
+        //revealing self
+        require(parameters.getState() == parameters.reveal(), "Not reveal state");
+        require(commitments[thisStakerId].epoch == epoch, "not commited in this epoch");
+        require(stakeManager.getStake(thisStakerId) > 0, "nonpositive stake");
+
+        votes[thisStakerId].epoch = epoch;
+        votes[thisStakerId].values = values;
+        uint256 influence = stakeManager.getInfluence(thisStakerId);
+        totalInfluenceRevealed[epoch] = totalInfluenceRevealed[epoch] + influence;
+        for (uint8 i = 0; i < values.length; i++) {
+            voteWeights[epoch][i][values[i]] = voteWeights[epoch][i][values[i]] + influence;
```

Comment:
By removing assetid from totalInfluenceRevealed, 
We consider that totalInflunceRevealed would be the same for assets
which is okay if all assets are revealed by every staker.

Issue happens if any staker reveals only subset of assets.
Here for the assets which were not revealed by the given staker, we will have same totalInfluenceRevealed.
Which would revert
finaliseDispute for all complement assets of the above subset.
As accWeight won't match totalInfluenceRevealed

https://github.com/razor-network/contracts/blob/dbd8bb2b248f7304948ae42a3e50feb9badd3ec4/contracts/Core/BlockManager.sol#L173

We can solve this by adding require in reveal, that staker should reveal all assets

Reply:
I think this can be fixed in my PR where i can make sure that the length of the array "votes" is equal to the number of active assets.

Reply:
Fixed in latest commit